src/Umbraco.Infrastructure/Persistence/SqlSyntax/SqlSyntaxProviderBase.cs

```diff
@@ -245,14 +264,16 @@ public virtual IEnumerable<ColumnInfo> GetColumnsInSchema(IDatabase db)
 
         public abstract void ReadLock(IDatabase db, params int[] lockIds);
         public abstract void WriteLock(IDatabase db, params int[] lockIds);
+
+        public virtual string GetFieldNameForUpdate<TDto>(Expression<Func<TDto, object>> fieldSelector, string tableAlias = null) => this.GetFieldName(fieldSelector, tableAlias);
+
+        public virtual IDictionary<Type, IScalarMapper> ScalarMappers => null;
+
         public abstract void ReadLock(IDatabase db, TimeSpan timeout, int lockId);
 
         public abstract void WriteLock(IDatabase db, TimeSpan timeout, int lockId);
```

Comment:
Reminder
Lock timeouts will happen when attempting to begin transaction.
Umbraco code mostly assumes transaction begin immediately but time out.